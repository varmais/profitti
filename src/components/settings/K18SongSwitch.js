import React, { Component, PropTypes } from 'react';
import {
  PixelRatio,
  StyleSheet,
  Switch,
  Text,
  View
} from 'react-native';
import { connect } from 'react-redux';
import config from '../../config';
import { setK18SettingValue } from '../../redux/settings';

const styles = StyleSheet.create({
  container: {
    height: 60,
    backgroundColor: config.colors.white,
    borderWidth: 0,
    justifyContent: 'center'
  },
  text: {
    color: config.colors.blackLight,
    fontSize: 18
  },
  switch: {
    position: 'absolute',
    right: 0,
    bottom: 27 / PixelRatio.get()
  }
});

export class K18SongSwitch extends Component {
  static propTypes = {
    k18Enabled: PropTypes.bool.isRequired,
    handleValueChange: PropTypes.func.isRequired
  };

  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>K18-laulut</Text>
        <Switch
          onTintColor={config.colors.green}
          tintColor={config.colors.greenlight}
          onValueChange={this.props.handleValueChange}
          value={this.props.k18Enabled}
          style={styles.switch}
        />
      </View>
    )
  }
}

export default connect((state) => ({
  k18Enabled: state.settings.k18Enabled
}), {
  handleValueChange: setK18SettingValue
})(K18SongSwitch);
