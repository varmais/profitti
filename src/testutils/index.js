import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

global.React = React;
global.renderer = renderer;

export function createNavigationProp (params = {}) {
  return {
    state: {params: params},
    navigate: () => {}
  }
}