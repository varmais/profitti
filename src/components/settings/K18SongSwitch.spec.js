import '../../testutils';
import { K18SongSwitch } from './K18SongSwitch';

describe('components:settings:K18SongSwitch', () => {
  let handleValueChangeStub;

  beforeEach(() => {
    handleValueChangeStub = jest.fn();
  });

  describe('when k18 is disabled', () => {
    test('renders K18SongSwitch', () => {
      const snapshot = renderer.create(<K18SongSwitch k18Enabled={false} handleValueChange={handleValueChangeStub} />).toJSON();
      expect(snapshot).toMatchSnapshot();
    });

    describe('when switch is pressed', () => {
      test('it calls onValueChange callback');
    });
  });

  describe('when k18 is enabled', () => {
    test('renders K18SongSwitch', () => {
      const snapshot = renderer.create(<K18SongSwitch k18Enabled={true} handleValueChange={handleValueChangeStub} />).toJSON();
      expect(snapshot).toMatchSnapshot();
    });
  });
});
