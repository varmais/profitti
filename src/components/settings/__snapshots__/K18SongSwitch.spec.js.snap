// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:settings:K18SongSwitch when k18 is disabled renders K18SongSwitch 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderWidth": 0,
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#222",
        "fontSize": 18,
      }
    }
  >
    K18-laulut
  </Text>
  <RCTSwitch
    disabled={false}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#9EB741"
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "bottom": 13.5,
          "position": "absolute",
          "right": 0,
        },
      ]
    }
    tintColor="#B7C83B"
    value={false}
  />
</View>
`;

exports[`components:settings:K18SongSwitch when k18 is enabled renders K18SongSwitch 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderWidth": 0,
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#222",
        "fontSize": 18,
      }
    }
  >
    K18-laulut
  </Text>
  <RCTSwitch
    disabled={false}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#9EB741"
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "bottom": 13.5,
          "position": "absolute",
          "right": 0,
        },
      ]
    }
    tintColor="#B7C83B"
    value={true}
  />
</View>
`;
